app/src/main/java/com/tvtracker/service/MediaService.kt

```diff
@@ -14,9 +14,8 @@ class MediaService : IMediaService {
     override suspend fun searchIMDB(text: String, type: String, page: Int): ImdbResponse? {
         return withContext(Dispatchers.IO) {
             val service = ImdbRetrofitClientInstance.retrofitInstance?.create(IMediaItemDAO::class.java)
-            val mediaItems = async {service?.searchIMDB(text, type, page)}
-            var result = mediaItems.await()?.awaitResponse()?.body()
-            return@withContext result
+            val mediaItems = async { service?.searchIMDB(text, type, page) }
+            return@withContext mediaItems.await()?.awaitResponse()?.body()
```

Comment:
The result from "mediaItems.await()?.awaitResponse()?.body()" could be returned in the same line, that's why I deleted the line where you were storing the body in a result variable.

Reply:
I caution against jamming too much into one line.  If you get an exception, with that line number listed, it's hard to figure out what caused the error.